great_expectations/data_context/data_context.py

```diff
@@ -110,83 +93,124 @@ def create(cls, project_root_dir=None):
 
         return cls(os.path.join(project_root_dir, "great_expectations"))
             
-    def __init__(self, context_root_dir=None, expectation_explorer=False, data_asset_name_delimiter='/'):
+    def __init__(self, project_config, context_root_dir, data_asset_name_delimiter='/'):
         """DataContext constructor
 
         Args:
             context_root_dir: location to look for the ``great_expectations.yml`` file. If None, searches for the file \
```

Comment:
Not sure what you're asking.

All of these changes are intended for `ConfigOnlyDataContex`t:
* context_root_dir is no longer optional.
* project_config is now specified at initialization, rather than loaded later from disc. This makes the base class more testable, since fixtures don't have to read/write from/to disc.
* expectation_explorer now belongs in a class of its own.

The inherited class `DataContext` retains the original behavior of DataContexts.

Broadly speaking, the change is "ConfigOnlyDataContexts don't guess about types or existence. Also, they don't attempt to keep their in-memory project_config in sync with an on-disc copy." Since ConfigOnlyDataContexts don't guess and have a single source of (in-memory) truth, they're a zillion times (actual number!) better for  testing.

Reply:
The Args: and the parameters are different, and I'm also still unsure at this point how to understand config-only given that it prominently takes a context_root_dir (but the Args is almost certainly now wrong since it says this is where the config search will happen).